src/main/java/co/cask/bigquery/BigQuerySink.java

```diff
@@ -126,7 +126,7 @@ public void prepareRun(BatchSinkContext context) throws Exception {
 
       String typeName = "STRING";
       if (type.isSimpleType()) {
-        typeName = simpleType(field.getSchema());
+        typeName = getTableDataType(field.getSchema());
       } else if (type == Schema.Type.UNION) {
```

Comment:
this is completely the wrong way to handle unions... it is assuming everything is the first thing in the union.... if NULL comes first it will treat it as null...